### ISSUE TITLE
No backward implementation for `torch._native_multi_head_attention`

### ISSUE THREAD

**Issue by lms-mt** [pytorch/pytorch#103484]
### 🚀 The feature, motivation and pitch

There is a forward implementation for `torch._native_multi_head_attention` but no coressponded backward implementation. So use torch to realize training MHA, we need to use small ops to compose it or use `torch.nn.MultiHeadAttention`. 

### Alternatives

* Small ops like linear/bmm to compose MHA.
* `torch.nn.MultiHeadAttention`

**Comment by mikaylagawarecki** [comment]
`_native_multi_head_attention` is a private function and should not be used, why do you want to use this instead of torch.nn.MultiHeadAttention?

**Comment by lms-mt** [comment]
> `_native_multi_head_attention` is a private function (as indicated by the leading underscore) and should not be used, why do you want to use this instead of torch.nn.MultiHeadAttention?

Yes I know. `torch.nn.MultiheadAttention` still has opportunity to goes into `_native_multi_head_attention` when it is in eval mode.
What I am conerned is that I want to adapt cuotomize device into Torch. So I was modifying the torch source code. I found `_native_multi_head_attention` 's function declaration corresponds to my expectation. So I will bind dispatch this func to device function. But I found it has no backward implementation and I want to implement it. But I choose to ask here, want to know why the backward func doesn't exist. 
This API seems good because I contains all the MHA operation. When it supoorts both forward and backward, I can implement my own MHA like `foo.MultiHeadAttention(nn.Module` and call it in our own scripts.

Glad to have your reply. @mikaylagawarecki
